lib/elixir/pages/anti-patterns/design-anti-patterns.md

```diff
@@ -0,0 +1,40 @@
+# Design-related anti-patterns
+
+This document outlines anti-patterns related to your modules, functions, and the role they
+play within a codebase.
+
+## Primitive obsession
+
+TODO
+
+## Boolean obsession
+
+TODO
+
+## Working with invalid data
+
+TODO
+
+## Alternative return types
+
+TODO
+
+## Unrelated multi-clause function
+
+TODO
+
+## Feature envy
+
+TODO
+
+## Excessive side-effects
+
+TODO (note: validate name)
```

Comment:
@lucasvegi this is a new anti-pattern about doing side-effects (writing to the database table, writing to IO) in the middle of functions. Some side-effect is fine but splitting it may be better. If you have suggestions on the name, we are glad to hear them!

Reply:
@josevalim This is an interesting smell. From your description, I understand that it is related to a function being VERY impure. How about the name "**Dirty function**"? In some way, I believe we should incorporate the fact that the function is highly impure into the name of the smell. A more traditional and analogy-free option could be "**Excessively Impure Function**".

Reply:
Thank you. Originally the name was about "purity" but we decided to focus on side-effects because I believe this is an easier to grasp concept if you don't come from a strict functional programming background.

Reply:
> Thank you. Originally the name was about "purity" but we decided to focus on side-effects because I believe this is an easier to grasp concept if you don't come from a strict functional programming background.

I understood. I agree with you

Reply:
@lucasvegi so, to confirm, are you happy with the proposed name?

Reply:
@josevalim Yes, I think he conveys the essence of the problem very well. I am satisfied!

If you find it convenient, in the description of this smell we can say something about the excess of impurity causing many side-effects, otherwise, we can ignore the concept of pure/impure function without problem too.